objects/src/accounts/code/mod.rs

```diff
@@ -42,36 +42,76 @@ impl AccountCode {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a new [AccountCode] instantiated from the provided [Library].
+    /// Creates a new [`AccountCode`] from the provided components' libraries.
     ///
-    /// All procedures exported from the provided library will become members of the account's
-    /// public interface.
+    /// # Warning
     ///
-    /// # Notes
-    /// - A `faucet` flag has been temporarly added to this procedure enabling correct storage
-    ///   offset of faucet procedures. This is needed because of the faucet reserved storage slot at
-    ///   location 0.
+    /// This does not check whether the provided components are valid when combined. Prefer
+    /// [`Account::initialize_from_components`](crate::accounts::Account::initialize_from_components)
+    /// whenever possible as it implements that check.
```

Comment:
It is just to avoid doing this check twice when calling it from `Account::initialize_from_components`, as that function does this check already.

I think a good approach would be to:
- Add a `pub(super) fn AccountCode::from_components_inner` function that does what the current function does and is called from `Account::initialize_from_components`.
- Add a `pub fn AccountCode::from_components` function that includes the validation check and otherwise delegates to the inner function. It could be behind the `testing` feature as it is only used in tests code.